app/src/main/java/org/p2p/wallet/striga/offramp/mappers/StrigaOffRampSwapWidgetMapper.kt

```diff
@@ -0,0 +1,121 @@
+import org.p2p.wallet.striga.offramp.models.StrigaOffRampTokenType
+
+class StrigaOffRampSwapWidgetMapper {
+
+    fun mapByState(tokenType: StrigaOffRampTokenType, state: StrigaOffRampTokenState): SwapWidgetModel {
+        return when (tokenType) {
+            StrigaOffRampTokenType.TokenA -> {
+                mapStateToSwapWidgetModel(
+                    titleResId = tokenType.titleResId,
+                    currencyName = tokenType.currencyName,
+                    amount = state.amount,
+                    balance = state.balance,
+                    isLoadingAmount = state is StrigaOffRampTokenState.Loading,
+                    isLoadingBalance = state is StrigaOffRampTokenState.LoadingBalance,
+                    isStatic = state is StrigaOffRampTokenState.Disabled
+                )
+            }
+            StrigaOffRampTokenType.TokenB -> {
+                mapStateToSwapWidgetModel(
+                    titleResId = tokenType.titleResId,
+                    currencyName = tokenType.currencyName,
+                    amount = state.amount,
+                    balance = null,
+                    isLoadingAmount = state is StrigaOffRampTokenState.Loading,
+                    isLoadingBalance = state is StrigaOffRampTokenState.LoadingBalance,
+                    isStatic = state is StrigaOffRampTokenState.Disabled
+                )
+            }
+        }
+    }
+
+    fun mapTokenA(
+        amount: BigDecimal,
+        balance: BigDecimal
+    ): StrigaOffRampTokenState {
+        return StrigaOffRampTokenState.Content(
+            amount = amount,
+            balance = balance
+        )
+    }
+
+    fun mapTokenALoadingBalance(amountA: BigDecimal): StrigaOffRampTokenState {
+        return StrigaOffRampTokenState.LoadingBalance(
+            amountA = amountA,
+        )
+    }
+
+    fun mapTokenB(amount: BigDecimal): StrigaOffRampTokenState = StrigaOffRampTokenState.Content(amount)
+
+    private fun mapStateToSwapWidgetModel(
+        titleResId: Int,
+        currencyName: String,
+        amount: BigDecimal,
+        balance: BigDecimal? = null,
+        isLoadingAmount: Boolean = false,
+        isLoadingBalance: Boolean = false,
+        isStatic: Boolean = false
+    ): SwapWidgetModel {
+        return SwapWidgetModel.Content(
+            isStatic = isStatic,
+            widgetTitle = mapWidgetTitle(titleResId),
+            availableAmount = if (isLoadingBalance) {
+                // todo: maybe we should use skeleton for availableAmount too?
+                // SwapWidget currently doesn't support skeleton for availableAmount
+                mapTokenAmountText(BigDecimal.ZERO)
+            } else {
+                // the "balance" is inconsistent with the "availableAmount" but it's clearly describes what it is,
+                // and of course the "balance" is shorter than the "availableAmount"
+                balance?.let { mapTokenAmountText(it) }
+            },
```

Comment:
how about extracting this logic with calculating availableAmount to put todo comments there? 
construction of Content will be much clearer